EvtWeightTools/src/Data_to_MC_CorrectionInterface_0l_4tau_trigger.cc

```diff
@@ -40,132 +40,30 @@ Data_to_MC_CorrectionInterface_0l_4tau_trigger::getSF_triggerEff(TriggerSFsys ce
     sys = "down";
   }
 
-  double eff_2tau_tauLeg1_data = 0.;
-  double eff_2tau_tauLeg1_mc   = 0.;
-  double eff_2tau_tauLeg2_data = 0.;
-  double eff_2tau_tauLeg2_mc   = 0.;
-  double eff_2tau_tauLeg3_data = 0.;
-  double eff_2tau_tauLeg3_mc   = 0.;
-  double eff_2tau_tauLeg4_data = 0.;
-  double eff_2tau_tauLeg4_mc   = 0.;
-
-  if(std::fabs(hadTau1_eta_) <= 2.1 && aux::hasDecayMode(allowedDecayModes_, hadTau1_decayMode_))
-  {
-    eff_2tau_tauLeg1_data = tau_leg_efficiency(hadTau1_pt_, hadTau1_decayMode_, "ditau", wp_str_, "eff_data", sys);
-    eff_2tau_tauLeg1_mc   = tau_leg_efficiency(hadTau1_pt_, hadTau1_decayMode_, "ditau", wp_str_, "eff_mc", sys);
-  }
-
-  if(std::fabs(hadTau2_eta_) <= 2.1 && aux::hasDecayMode(allowedDecayModes_, hadTau2_decayMode_))
-  {
-    eff_2tau_tauLeg2_data = tau_leg_efficiency(hadTau2_pt_, hadTau2_decayMode_, "ditau", wp_str_, "eff_data", sys);
-    eff_2tau_tauLeg2_mc   = tau_leg_efficiency(hadTau2_pt_, hadTau2_decayMode_, "ditau", wp_str_, "eff_mc", sys);
-  }
-
-  if(std::fabs(hadTau3_eta_) <= 2.1 && aux::hasDecayMode(allowedDecayModes_, hadTau3_decayMode_))
-  {
-    eff_2tau_tauLeg3_data = tau_leg_efficiency(hadTau3_pt_, hadTau2_decayMode_, "ditau", wp_str_, "eff_data", sys);
-    eff_2tau_tauLeg3_mc   = tau_leg_efficiency(hadTau3_pt_, hadTau3_decayMode_, "ditau", wp_str_, "eff_mc", sys);
-  }
+  std::vector<const RecoHadTau*> matched_hadTaus;
 
-  if(std::fabs(hadTau4_eta_) <= 2.1 && aux::hasDecayMode(allowedDecayModes_, hadTau4_decayMode_))
+  for (auto hadTau_recObj : hadTau_recObjs)
```

Comment:
I guess removing the check on abs(eta) of the hadronic tau can be justified by requiring it to be matched to a di-tau trigger object which needs to be in the acceptance region anyways, but a comment could help explaining this.

Reply:
Ah ok, I see that `setHadTau` already checks the eligibility of hadronic taus, so ignore this comment.